### PR TITLE
ipv6_addr: do not null remaining bits in prefix initialization

### PR DESCRIPTION
**Pull request by authmillenon**
To override the prefix of an already existing address seems to be the more common use-case, so I changed the semantics of this function. The previous semantics can still be achieved using

```C
ng_ipv6_addr_set_unspecified(addr);
ng_ipv6_addr_init_prefix(addr, prefix, prefix_len);
```

### CODE REVIEW

File: sys/net/network_layer/ng_ipv6/addr/ng_ipv6_addr.c [hunk 1]
```diff
@@ -75,7 +73,7 @@ void ng_ipv6_addr_init_prefix(ng_ipv6_addr_t *out, const ng_ipv6_addr_t *prefix,
     if (bits % 8) {
         uint8_t mask = 0xff << (8 - (bits - (bytes * 8)));
 
-        out->u8[bytes] = prefix->u8[bytes] & mask;
+        out->u8[bytes] |= prefix->u8[bytes] & mask;
```
Comment (Kijewski) [inline sys/net/network_layer/ng_ipv6/addr/ng_ipv6_addr.c]: This is a no op. :) Probably you meant `out->u8[bytes] &= mask` or `out->u8[bytes] &= ~mask`.

Comment (authmillenon) [inline sys/net/network_layer/ng_ipv6/addr/ng_ipv6_addr.c]: I meant both: I want to override the prefix after all ;)

**Comment by OlegHahm** [comment]
ACK

**Comment by authmillenon** [comment]
Squashed

**Comment by authmillenon** [comment]
Oh, and adapted unittests ;-)

**Pull request merged**